packages/portal-contracts/test/erc721Gateway.test.ts

```diff
@@ -0,0 +1,1044 @@
+          .connect(env.signers[1])
+          .grantRole(defaultAdminRole, signer2)
+      ).to.be.revertedWith(
+        `AccessControl: account ${env.addresses[1].toLowerCase()} is missing role ${defaultAdminRole}`
+      );
+      expect(
+        await env.fuelERC721Gateway.hasRole(defaultAdminRole, signer2)
+      ).to.equal(false);
+
+      // Attempt grant pauser role
+      await expect(
+        env.fuelERC721Gateway
+          .connect(env.signers[1])
+          .grantRole(pauserRole, signer2)
+      ).to.be.revertedWith(
+        `AccessControl: account ${env.addresses[1].toLowerCase()} is missing role ${defaultAdminRole}`
+      );
+      expect(await env.fuelERC721Gateway.hasRole(pauserRole, signer2)).to.equal(
+        false
+      );
+    });
+
+    it('Should be able to revoke pauser role', async () => {
+      expect(await env.fuelERC721Gateway.hasRole(pauserRole, signer1)).to.equal(
+        true
+      );
+
+      // Grant pauser role
+      await expect(env.fuelERC721Gateway.revokeRole(pauserRole, signer1)).to.not
+        .be.reverted;
+      expect(await env.fuelERC721Gateway.hasRole(pauserRole, signer1)).to.equal(
+        false
+      );
+    });
+  });
+
+  describe('Make both valid and invalid ERC721 deposits', async () => {
+    let provider: Provider;
+    before(async () => {
+      provider = env.fuelMessagePortal.provider;
+    });
+
+    it('Should be able to deposit tokens', async () => {
+      const testDeposit = async (
+        signer: SignerWithAddress,
+        fuelTarget: string,
+        token: NFT,
+        tokenId: BigNumberish
+      ) => {
+        const toAddress = randomBytes32();
+
+        const expectedMessageData = computeMessageData(
+          fuelTarget,
+          tokenAddress.split('0x').join('0x000000000000000000000000'),
+          tokenId,
+          signer.address.split('0x').join('0x000000000000000000000000'),
+          toAddress,
+          1
+        );
+        const expectedNonce =
+          await env.fuelMessagePortal.getNextOutgoingMessageNonce();
+        const expectedAmount = 0;
+
+        await env.nft
+          .connect(signer)
+          .approve(env.fuelERC721Gateway.address, tokenId);
+        await expect(
+          env.fuelERC721Gateway
+            .connect(signer)
+            .deposit(toAddress, token.address, fuelTarget, tokenId)
+        )
+          .to.emit(env.fuelMessagePortal, 'MessageSent')
+          .withArgs(
+            gatewayAddress,
+            CONTRACT_MESSAGE_PREDICATE,
+            expectedNonce,
+            expectedAmount,
+            expectedMessageData
+          );
+
+        expect(await env.nft.ownerOf(tokenId)).to.be.equal(
+          env.fuelERC721Gateway.address
+        );
+
+        expect(
+          await env.fuelERC721Gateway.tokensDeposited(env.nft.address, tokenId)
+        ).to.be.equal(fuelTarget);
+      };
+
+      // Deposit to fuelTokenTarget1
+      {
+        const tokenId = 0;
+        await testDeposit(
```

Comment:
Should these have assertions?

Reply:
It 'd be redundant. `ethers` throws when the transaction is not successful